src/web/routes/application/flow-control/middleware/handle-path-request/handle-path-request.js

```diff
@@ -3,27 +3,21 @@ const { stateMachine, actions, states } = require('../../state-machine')
 const { stepNotNavigable } = require('./predicates')
 
 const { COMPLETED } = states
-const { IS_PATH_ALLOWED, GET_NEXT_ALLOWED_PATH, SET_NEXT_ALLOWED_PATH } = actions
+const { IS_PATH_ALLOWED, GET_NEXT_ALLOWED_PATH } = actions
 
+// TODO HTBHF-2506 remove config from signature as no longer required
 const handleRequestForPath = (config, journey, step) => (req, res, next) => {
   const { pathsInSequence } = journey
   const firstPathInSequence = pathsInSequence[0]
 
   // Destroy the session on navigating away from CONFIRM_URL
-  if (stateMachine.getState(req) === COMPLETED && req.path !== CONFIRM_URL) {
+  if (stateMachine.getState(req, journey) === COMPLETED && req.path !== CONFIRM_URL) {
     req.session.destroy()
     res.clearCookie('lang')
     return res.redirect(firstPathInSequence)
   }
 
-  // Initialise nextAllowedPath if none exists in session
-  let nextAllowedPath = stateMachine.dispatch(GET_NEXT_ALLOWED_PATH, req, journey)
-
-  if (!nextAllowedPath) {
-    stateMachine.dispatch(SET_NEXT_ALLOWED_PATH, req, journey, firstPathInSequence)
-    nextAllowedPath = firstPathInSequence
-  }
-
+  const nextAllowedPath = stateMachine.dispatch(GET_NEXT_ALLOWED_PATH, req, journey)
   const isPathAllowed = stateMachine.dispatch(IS_PATH_ALLOWED, req, journey)
```

Comment:
This has been removed as the session is now initialised with the first path in the journey sequence, so `GET_NEXT_ALLOWED_PATH` will always return the correct path.